GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -905,17 +978,84 @@ public async Task<(decimal purchasePrice, decimal customerPrice, decimal fromPri
             return result;
         }
 
-        /// <summary>
-        /// Get the prices from an API if the API is setup for the configurator.
-        /// </summary>
-        /// <param name="configuration">The configuration to get the price for.</param>
-        /// <param name="dataTable">The data table with the configurator data.</param>
-        /// <returns>Returns a tuple with the three prices as decimal.</returns>
-        private async Task<(decimal purchasePrice, decimal customerPrice, decimal fromPrice)> GetPriceFromApiAsync(ConfigurationsModel configuration, DataTable dataTable)
+        /// <inheritdoc />
+        public async Task<(decimal purchasePrice, decimal customerPrice, decimal fromPrice)> CalculatePriceAsync(VueConfigurationsModel input)
         {
             (decimal purchasePrice, decimal customerPrice, decimal fromPrice) result = (0, 0, 0);
 
-            var configuratorId = Convert.ToUInt64(dataTable.Rows[0].Field<object>("configuratorId"));
+            var configuratorData = await GetVueConfiguratorDataAsync(input.ConfiguratorName, false);
+            if (configuratorData == null)
+            {
+                return result;
+            }
+
+            try
+            {
+                // Get the price from an API if available. If not it will return 0, 0, 0.
+                var priceFromApi = await GetPriceFromApiAsync(configuratorData.ConfiguratorId, vueConfiguration: input);
+                result.purchasePrice += priceFromApi.purchasePrice;
+                result.customerPrice += priceFromApi.customerPrice;
+                result.fromPrice += priceFromApi.fromPrice;
+            }
+            // If an exception is thrown during the retrieval of the price from an API consider the full price to be invalid.
+            catch (ArgumentException e)
+            {
+                // ArgumentException is thrown when the response of the API was not successful.
+                logger.LogError(e, "Error while trying to get price from an API.");
+                return result;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "Error while trying to get price from an API.");
+                return result;
+            }
```

Comment:
You handle 2 different exceptions the exact same way here. The catch `ArgumentException` is not very useful now, if you remove that entire catch, you will get the exact same results.